### ISSUE TITLE
Change HTTPHeader on custom

### ISSUE THREAD

**Issue by SpectralDragon** [p2/OAuth2#125]
Hi, my problem is that I need to put down in the header `application/json, text/plain, */*`, instead of `application/json`, a request to transfer the body. When the log option, the server responds with 406, and if I am directly `OAuth2AuthRequest.asURLRequestFor(oauth2:)`, then everything is fine refunded. Is there a way to change the "header" without change in the method of directly `OAuth2AuthRequest`?

**Comment by p2** [comment]
You can directly manipulate the request you're getting:

```swift
let req = oauth2.request(forURL: <# url #>)
req.setValue("application/json, text/plain, */*", forHTTPHeaderField: "Accept")

let task = oauth2.session.dataTaskWithRequest(req) { data, response, error in
    ...
}
task.resume()
```

Does this help?

**Comment by SpectralDragon** [comment]
@p2  No, because this header I need for method `authorize`.

**Comment by p2** [comment]
Ah I see. Are you using a code grant flow? In this case you would need to subclass `OAuth2CodeGrant` and override [`exchangeCodeForToken()`](https://github.com/p2/OAuth2/blob/master/Sources/Base/OAuth2CodeGrant.swift#L88) to change the header. Copy/paste the complete original method, then change the header on `post` after this line:

    let post = try tokenRequestWithCode(code).asURLRequestFor(self)

I will make this easier for version 3 of the library, which lives on the develop branch. Let me know if you're using the develop branch and wish to test.

**Comment by p2** [comment]
On the `develop` branch, you can now achieve this by subclassing `OAuth2CodeGrant` like so:

```swift
open class OAuth2CodeGrantCustom: OAuth2CodeGrant {
    override open func accessTokenRequest(with code: String, params: OAuth2StringDict? = nil) throws -> OAuth2AuthRequest {
        let req = try super.accessTokenRequest(with: code, params: params)
        req.setHeader("Accept", to: "application/json, text/plain, */*")
        return req
    }
}
```

**Comment by p2** [comment]
This has been implemented by @SpectralDragon in version 2.2.9, custom auth headers can now be specified using the `headers` dictionary key or on `oauth2.authConfig.authHeaders`.
